R/derive_extreme_event.R

```diff
@@ -252,6 +380,37 @@
 #'     new_vars = exprs(TRTSDT)
 #'   )
 #'
+#' @info Since the CBOR derivation contains multiple complex parts, it's
+#'   convenient to make use of the `description` argument within each event object
+#'   to describe what condition is being checked.
+#'
+#'   - For the Confirmed Response (CR), for each `"CR"` record in the original ADRS
+#'     dataset that will be identified by the first part of the `condition` argument
+#'     (`AVALC == "CR"`) we need to use the `first_cond_upper` argument to limit the
+#'     group of observations to consider alongside it. Namely, we need to look up to
+#'     and including the second CR (`AVALC.join == "CR"`) over 28 days from the first
+#'     one (`ADT.join >= ADT + 28`). The observations satisfying `first_cond_upper`
+#'     then form part of our "join group", meaning that the remaining portions of
+#'     `condition` which reference joined variables are limited to this group.
+#'     In particular, within `condition` we use `all()` to check that all observations
+#'     are either `"CR"` or `"NE"`, and `count_vals()` to ensure at most one is
+#'     `"NE"`. Note that here we have selected `join_type = "after"` because
+#'     <insert explanation here, explaining why join_type = "all" changes patient 1 to a "PR">.
+#'
+#'  - The Partial Response (PR), is very similar; with the difference being that the
+#'    first portion of `condition` now references `"PR"` and `first_cond_upper`
+#'    accepts a confirmatory PR or CR 28 days later. Note that now we must add
+#'    `"PR"` as an option within the `all()` condition to account for confirmatory
+#'    PRs.
+#'
+#'  - The Stable Disease (SD), Progressive Disease (PD) and Not Evaluable (NE)
+#'    events are simpler and just require `event()` calls.
+#'
+#'  - Finally, we use a catch-all `event()` with `condition = TRUE` and
+#'    `dataset_name = adsl` to identify those subjects who do not appear in ADRS
```

Comment:
Explain you also need `source_datasets = list(adsl = adsl)` in the function call as you're using a different input dataset to `dataset` for this `event()`.

Reply:
agree, will add.